### PR TITLE
[objectives.lua] improve clarity of error message

### PR DESCRIPTION
**Pull request by knyghtmare**
Current message:
```
Unknown Condition, ignoring
```
This really does not tell me anything of where I might have messed up and can be even more confusing if I have custom lua from the campaign or an add-on being used in the same scenario.

The edited one tells me where I actually have to look for the error message and fix it.

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
I don't really understand why it's `wml.interface.add_chat_message` instead of either `wml.error` or `warn`… but if we're _insisting_ on using `add_chat_message`, we should probably explicitly set the speaker to `"WML"` instead of leaving it at the default, which is `"Lua"`.

**Comment by soliton-** [comment]
I see no reason why this should not use `wml.error`.

File: data/lua/wml/objectives.lua [hunk 1]
```diff
@@ -87,7 +87,7 @@ local function generate_objectives(cfg)
 					.. color_prefix(r, g, b) .. objective_bullet.. description
 					.. turn_counter .. "</span>" .. "\n"
 			else
-				wesnoth.interface.add_chat_message "Unknown condition, ignoring."
+				wesnoth.interface.add_chat_message "[objective] Either condition= is missing or condition= has invalid value (ignoring)."
```
Comment (soliton-) [inline data/lua/wml/objectives.lua:90]: I'd include the value of `condition` in the message.

**Comment by knyghtmare** [comment]
> I don't really understand why it's wml.interface.add_chat_message instead of either wml.error

I got no clue either. I was not the one who wrote this code initially.

> I see no reason why this should not use wml.error.

Done.

> I'd include the value of condition in the message.

I know how to do this in python (my foundation) but who does one do this in Lua?

**Comment by CelticMinstrel** [comment]
> I know how to do this in python (my foundation) but who does one do this in Lua?

You can use either `string.format` (printf-style) or `stringx.vformat` (WML variable substitution style). I'd recommend the latter.

**Comment by CelticMinstrel** [comment]
I still say I'd prefer the value to be added to the message. Are you planning to add that yourself?

**Comment by Pentarctagon** [comment]
I wasn't, but it seemed like an improvement over the current error message regardless.

**Pull request merged**